### PR TITLE
feat: 增加允许/禁用指标在行列维度中相互切换的能力

### LINKED ISSUE #1544: 🙏 Switcher支持通过配置禁用行列切换和disable特定区域
### 🖋 Description
遇到一些场景需要禁用掉行头 和 列头的切换功能
image
image

希望能新增以下配置
- 可以禁用掉行列之间的切换  
- 各个区域自身可以配置是否disable（样式disable/不可选择/拖拽顺序）（优先级低）

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [x] New feature, close #1544 

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| 开启 | 关闭 |
| ------ | ----- |
|       |  |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: s2-site/docs/api/components/switcher.zh.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ order: 3
 | contentTitleText | 弹窗默认标题文字 | `string` | 行列切换 |  |
 | resetText | 重置按钮文字 | `string` | 恢复默认 |  |
 | innerContentClassName | 弹出框内容样式名 | `string` |  |  |
+| allowSwitchBetweenRowsAndCols| 是否允许指标在行列维度之间相互切换 | `boolean` |  |  `true`|
```
Comment (xingwanying) [inline s2-site/docs/api/components/switcher.zh.md:20]: 这个名字，是不是有些过长了🤔

Comment (wjgogogo) [reply s2-site/docs/api/components/switcher.zh.md:20]: 就是期待大家能给一个更好的命名

Comment (lijinke666) [reply s2-site/docs/api/components/switcher.zh.md:20]: allowExchangeHeader ?

Comment (xingwanying) [reply s2-site/docs/api/components/switcher.zh.md:20]: enableDimensionSwitcher

Comment (wjgogogo) [reply s2-site/docs/api/components/switcher.zh.md:20]: @lijinke666 你这个要好点
@xingwanying dimension这个词可能对于DI来说比较熟悉，毕竟row是dimension,col是dimensionY，但是对于外部同学来说，dimension对应行列头的关系就没有那么明显

**Review by xingwanying — APPROVED** [review]
LG

**Pull request merged**